provisioner/aws-manager.js

```diff
@@ -369,84 +317,48 @@ AwsManager.prototype.listSpotRequestIds = function() {
  */
 AwsManager.prototype.capacityForType = function(workerType, states) {
   assert(workerType);
-  var that = this;
-  var wName = workerType.workerType;
-  var capacity = 0;
   if (!states) {
     states = ['running', 'pending', 'spotReq'];
   }
+  var wName = workerType.workerType;
+  var capacity = 0;
+  var instances = this.instancesOfType(workerType.workerType);
+  var requests = this.requestsOfType(workerType.workerType);
+  // TODO: Make sure internal state is tracked!
 
-  // Find instances in the retrevied state and add them to the capacity
-  // according to their declared capacity
-  this.ec2.regions.forEach(function(region) {
-    var rState = that.getApi(region);
-
-    if (!rState[wName]) {
-      return;
-    }
-
-    var wState = rState[wName];
-
-    /* When counting capacity of each type, if we have instances
-       of a type which is no longer in the workerType definition
-       we cannot know what it's capacity is.  We'll try to get its
-       real capacity, but fall back to 1 as a safe option */
-    if (states.includes('running')) {
-      wState.running.forEach(function(instance) {
-        try {
-          capacity += workerType.capacityOfType(instance.InstanceType);
-        } catch (err) {
-          capacity += 1;
-        }
-      });
-    }
 
-    if (states.includes('pending')) {
-      wState.pending.forEach(function(instance) {
-        try {
-          capacity += workerType.capacityOfType(instance.InstanceType);
-        } catch (err) {
-          capacity += 1;
-        }
-      });
+  instances.forEach(function(instance) {
+    if (states.includes(instance.State.Name)) {
+      try {
+        capacity += workerType.capacityOfType(instance.InstanceType);
+      } catch(err) {
+        capacity++;
+      }
```

Comment:
Another approach using lodash `_.sum` that I just discovered today:
```js
var instanceToCapacity = (instance) => {
  try {
     return workerType.capacityOfType(instance.InstanceType);
  }
  catch(err) {
    return 1;
  }
};

capacity += _.sum(
  instances.filter(instance => {
    return states.includes(instance.State.Name);
  }).map(instanceToCapacity)
);
```

Either way works... But I would consider moving error handling code into `workerType.capacityOfType`.
Or make sure that illegal instanceTypes are ignored and immediately murdered by rougeKiller (seems fair).